lib/ipmi_strings.c

```diff
@@ -1735,7 +1735,7 @@ int ipmi_oem_info_init()
 		goto out;
 	}
 
-	if (!(count = oem_info_list_load(&oemlist))) {
+	if ((count = oem_info_list_load(&oemlist)) < 1) {
```

Comment:
As I look at the contents of this conditional block now, I suspect it's actually not correct to set `ipmi_oem_info` to `ipmi_oem_dummy` in any case. Instead, `oem_info_init_from_list()` should be taught to properly build an info list (with head and tail) from an empty input.

Reply:
So you want to drop the `ipmi_oem_dummy` struct completely? If so, i'll work that into the reworked commits

Reply:
This has implications for `oem_info_init_from_list` because if it can't allocate memory, it uses the dummy list. If we get rid of it, then ipmitool will need to always be able to alloc at least `(head_entries + tail_entries + 1) * sizeof(*ipmi_oem_info)` bytes of memory instead of being able to rely on that static struct.

If that memory allocation fails, then `ipmi_oem_info_init` needs to remain non-void to propagate the failure.

Reply:
I think i understood your proposal. should be addressed